src/dstack/_internal/server/background/tasks/process_submitted_jobs.py

```diff
@@ -106,7 +107,15 @@ async def _process_submitted_job(session: AsyncSession, job_model: JobModel):
     )
 
     run = run_model_to_run(run_model)
-    job = run.jobs[job_model.job_num]
+    job = find_job(run.jobs, job_model.replica_num, job_model.job_num)
```

Comment:
It seems there is no need in find_job returning Optional. Maybe it should return value or error? Also, it seems like getting a job should always succeed, so there is no need to handle the None case explicitly.